go.mod

```diff
@@ -8,25 +8,17 @@ retract (
 	[v1.6.0, v1.6.1] // Contains data race bug in background connection establishment.
 )
 
-// gopkg.in/yaml.v3@v3.0.0-20200313102051-9f266ea9e77c introduced through github.com/stretchr/testify@v1.6.1 is
-// vulnerable to Denial of Service (DoS) via the Unmarshal function, which causes the program to crash when attempting
-// to deserialize invalid input. https://www.cve.org/CVERecord?id=CVE-2022-28948
-replace gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c => gopkg.in/yaml.v3 v3.0.1
-
 require (
-	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/davecgh/go-spew v1.1.1
 	github.com/golang/snappy v0.0.1
 	github.com/google/go-cmp v0.5.2
 	github.com/klauspost/compress v1.13.6
 	github.com/kr/pretty v0.1.0
 	github.com/montanaflynn/stats v0.0.0-20171201202039-1bf9dbcd8cbe
-	github.com/stretchr/testify v1.6.1
-	github.com/tidwall/pretty v1.0.0
+	github.com/tidwall/pretty v1.2.1
```

Comment:
`tidwall/pretty@1.0.0` imported `testify`, so I needed to update to a version that didn't use `testify` to get rid of `testify` and its dependencies entirely.

Reply:
This package is only used in a single place ([bson_corpus_spec_test.go](https://github.com/mongodb/mongo-go-driver/blob/eff5307d71ecbc763735076f3e2f63b5d77ead2a/bson/bson_corpus_spec_test.go)), is it possible to remove the dependency completely?

Reply:
We actually only use the [`Ugly`](https://github.com/tidwall/pretty/blob/master/pretty.go#L53) function which I was able to replace with `json.Compact`. Removed the `tidwall/pretty` dependency entirely.